js/ui/codeView.js

```diff
@@ -453,6 +456,44 @@ var CodingSession = new Lang.Class({
         return this._toolbox;
     },
 
+    setGrabbed: function(grabbed) {
+        if (this._button)
+            this._button.opacity = grabbed ? 0 : 255;
+
+        if (this.toolbox && this._state == STATE_APP)
+            this.toolbox.opacity = grabbed ? 0 : 255;
+    },
+
+    _attachAppToToolbox: function() {
+        this._appParent = this.app.get_parent();
+        this._appParent.remove_child(this.app);
+        this.toolbox.insert_child_at_index(this.app, 0);
+        this.app.set_position(0, 0);
+        this._appPositionId = this.app.connect('notify::position', (actor) => {
+            const [x, y] = actor.get_position();
+            if (x || y)
+                actor.set_position(0, 0);
```

Comment:
The reason for this seems a bit non-trivial (at least to me), can you explain (and if you agree it's not obvious, add a comment) why this is needed? Maybe to prevent the actor changing its position by itself when it's contained in the parent?

Reply:
We need this here because we're syncing the window position and after adding as a child we don't need to move the position of the child, so this is done to ensure that the child actor will be in the correct possition without modifying the sync function. I'll add a comment as you suggested